flyteplugins/go/tasks/logs/logging_utils.go

```diff
@@ -84,39 +85,45 @@ func (t templateLogPluginCollection) GetTaskLogs(input tasklog.Input) (tasklog.O
+			plugins = append(plugins, tasklog.TemplateLogPlugin{DisplayName: "Cloudwatch Logs", TemplateURIs: []tasklog.TemplateURI{fmt.Sprintf("https://console.aws.amazon.com/cloudwatch/home?region=%s#logEventViewer:group=%s;stream=var.log.containers.{{ .podName }}_{{ .namespace }}_{{ .containerName }}-{{ .containerId }}.log", cfg.CloudwatchRegion, cfg.CloudwatchLogGroup)}, MessageFormat: core.TaskLog_JSON})
 		}
 	}
 
 	if cfg.IsStackDriverEnabled {
 		if len(cfg.StackDriverTemplateURI) > 0 {
-			plugins = append(plugins, tasklog.TemplateLogPlugin{DisplayName: "Stackdriver Logs", Scheme: tasklog.TemplateSchemePod, TemplateURIs: []tasklog.TemplateURI{cfg.StackDriverTemplateURI}, MessageFormat: core.TaskLog_JSON})
+			plugins = append(plugins, tasklog.TemplateLogPlugin{DisplayName: "Stackdriver Logs", TemplateURIs: []tasklog.TemplateURI{cfg.StackDriverTemplateURI}, MessageFormat: core.TaskLog_JSON})
 		} else {
-			plugins = append(plugins, tasklog.TemplateLogPlugin{DisplayName: "Stackdriver Logs", Scheme: tasklog.TemplateSchemePod, TemplateURIs: []tasklog.TemplateURI{fmt.Sprintf("https://console.cloud.google.com/logs/viewer?project=%s&angularJsUrl=%%2Flogs%%2Fviewer%%3Fproject%%3D%s&resource=%s&advancedFilter=resource.labels.pod_name%%3D{{ .podName }}", cfg.GCPProjectName, cfg.GCPProjectName, cfg.StackdriverLogResourceName)}, MessageFormat: core.TaskLog_JSON})
+			plugins = append(plugins, tasklog.TemplateLogPlugin{DisplayName: "Stackdriver Logs", TemplateURIs: []tasklog.TemplateURI{fmt.Sprintf("https://console.cloud.google.com/logs/viewer?project=%s&angularJsUrl=%%2Flogs%%2Fviewer%%3Fproject%%3D%s&resource=%s&advancedFilter=resource.labels.pod_name%%3D{{ .podName }}", cfg.GCPProjectName, cfg.GCPProjectName, cfg.StackdriverLogResourceName)}, MessageFormat: core.TaskLog_JSON})
 		}
 	}
 
-	if cfg.IsFlyinEnabled {
-		if len(cfg.FlyinTemplateURI) > 0 {
-			plugins = append(plugins, tasklog.TemplateLogPlugin{DisplayName: "Flyin Logs", Scheme: tasklog.TemplateSchemeFlyin, TemplateURIs: []tasklog.TemplateURI{cfg.FlyinTemplateURI}, MessageFormat: core.TaskLog_JSON})
-		} else {
-			plugins = append(plugins, tasklog.TemplateLogPlugin{DisplayName: "Flyin Logs", Scheme: tasklog.TemplateSchemeFlyin, TemplateURIs: []tasklog.TemplateURI{fmt.Sprintf("https://flyin.%s/logs/{{ .namespace }}/{{ .podName }}/{{ .containerName }}/{{ .containerId }}", cfg.GCPProjectName)}, MessageFormat: core.TaskLog_JSON})
-		}
+	for logLinkType, dynamicLogLink := range cfg.DynamicLogLinks {
+		dynamicPlugins = append(
+			dynamicPlugins,
+			tasklog.TemplateLogPlugin{
+				Name:        logLinkType,
+				DisplayName: fmt.Sprintf("%s logs", logLinkType),
```

Comment:
nit: It won't always be "logs", like `vscode logs` isn't the right description, `wandb logs` maybe... right? I think just `logLinkType` is fine... alternatively, maybe the config should just be the logplugin?
so propeller config would be:
```
plugins:
  logs:
    dynamic-log-links:
      vscode: 
        displayName: "vscode debugger"
        templateURIs:
          - http://flyteinteractve.vs:{{ .taskConfig.port }}/{{ .podName }}
```

Reply:
pflags is not able to generate code for this. (I'm not familiar with the code, but from a cursory look it only parses [these](https://github.com/flyteorg/flyte/blob/master/flytestdlib/cli/pflags/api/generator.go#L30-L39) types).

If I define `LogConfig` like this:
```
type DynamicLogLink struct {
	TemplateURI tasklog.TemplateURI `json:"template-uri" pflag:",Template Uri to use when building log links"`
	DisplayName string              `json:"display-name" pflag:",Display name for the generated log when displayed in the console."`
}

// LogConfig encapsulates plugins' log configs
type LogConfig struct {
	...

	DynamicLogLinks map[string]*DynamicLogLink `json:"dynamic-log-links" pflag:",Map of dynamic log links"`

	Templates []tasklog.TemplateLogPlugin `json:"templates" pflag:"-,"`
}
```

I get [this](https://github.com/flyteorg/flyte/blob/master/flytestdlib/cli/pflags/api/generator.go#L102) error:
```
...
INFO[0000] [dynamic-log-links] is of a map type with default value []. 
Error: map of type [*github.com/flyteorg/flyte/flyteplugins/go/tasks/logs.DynamicLogLink] is not supported. Only basic slices or slices of json-unmarshalable types are supported
```

Reply:
1. Why not use TemplateLogPlugin :-D? 
2. Just exclude it from pflags:
    ```
	DynamicLogLinks map[string]*DynamicLogLink `json:"dynamic-log-links" pflag:"-,Map of dynamic log links"`
    ```